### PR TITLE
Add Rabbit's BS Viewer Presence

### LINKED ISSUE #2568: Rabbit's BS Viewer | skystudioapps.com/bs-viewer
**Prerequisites and essential questions** 
- [ ] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
Flowyan#1111
**What's the name of the service?** 
+1 Rabbit's BS Viewer | https://skystudioapps.com/bs-viewer/
**What should the Presence display?** 
The current map playing and its difficulty
**If possible, please provide a logo for the service (512x512 minimum)**

### PR DESCRIPTION
**Pull request by joerkig**
Add Presence for [Rabbit's BS Viewer](https://skystudioapps.com/bs-viewer/)
Resolves #2568
Discord_6wjJ7vy0Uy
Discord_j9Ys9ugFfP
Discord_mJqfsQBnqd
Discord_UfcErIuzBY
Discord_mfdjZH85EY
Discord_ld2yd3pKIh

### CODE REVIEW

File: websites/R/Rabbit's BS Viewer/presence.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  const elapsed = await presence.getSetting("elapsed"),
+    timeleft = await presence.getSetting("timeleft"),
+    buttons = await presence.getSetting("buttons"),
+    presenceData: PresenceData = {
+      largeImageKey: "logo",
+      startTimestamp: browsingStamp
+    };
+
+  if (document.location.pathname.includes("/bs-viewer/")) {
+    if (document.querySelector("#songname").textContent == "") {
```
Comment (ririxidev) [inline websites/R/Rabbit's BS Viewer/presence.ts:16]:
```suggestion
    if (document.querySelector("#songname").textContent === "") {
```
we always check type 😉

File: websites/R/Rabbit's BS Viewer/presence.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        + " ("
+        + (document.getElementById("difficultyselect") as HTMLSelectElement).selectedOptions.item(0).textContent
+        + ")";
+      presenceData.buttons = [
+        {
+          label: "View Page",
+          url: document.location.href
+        }
+      ];
+      if (document.location.href == "https://skystudioapps.com/bs-viewer/")
```
Comment (ririxidev) [inline websites/R/Rabbit's BS Viewer/presence.ts:36]:
```suggestion
      if (document.location.href === "https://skystudioapps.com/bs-viewer/")
```

**Review by ririxidev — COMMENTED** [review]
can't you do more in mapping-tools subpage?

**Review by joerkig — COMMENTED** [review]
Discord_qYja4q7HC0

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**